packages/domutils/tsconfig.json

```diff
@@ -1,12 +1,15 @@
 {
   "compilerOptions": {
     "composite": true,
+    "sourceMap": true,
     "declaration": true,
+    "declarationDir": "./types",
+    "declarationMap": true,
```

Comment:
What's the significance of moving the typings into a new separate dir? What exactly does `"declarationMap"` do here?

Reply:
This is a slightly opinionated change to avoid generating code into the same folder as the sources.  
It works as the package.json has a "types" property specifically to point to the new location.
`declarationMap` is analogous to `sourceMap` but for definition files.

Reply:
`declarationMap` is mostly useful for:
- when you are linking locally (e.g. yarn link). With it, if you try to follow a lumino type to its declaration from lab source code, and lumino is linked locally, you will go to the source file, not the typings file.
- if you distribute the TS source with the package, you will be able to do the above also for non-linked cases.